src/components/scene/background.js

```diff
@@ -1,49 +1,229 @@
+      }
+    });
+
+    this.el.sceneEl.addEventListener('exit-vr', function () {
+      self.stopLightProbe();
+    });
+
+    var directionalLight = document.createElement('a-light');
+    directionalLight.setAttribute('type', 'directional');
+    directionalLight.setAttribute('intensity', 0);
+    this.el.appendChild(directionalLight);
+
+    var probeLight = document.createElement('a-light');
+    probeLight.setAttribute('type', 'probe');
+    probeLight.setAttribute('intensity', 0);
+    this.el.appendChild(probeLight);
+
+    this.directionalLight = directionalLight;
+    this.probeLight = probeLight;
+    this.sceneLightsMap = new Map();
+  },
+  stopLightProbe: function () {
+    var self = this;
+    var data = this.data;
+    var scene = this.el.sceneEl.object3D;
+    this.xrLightProbe = null;
+
+    var sceneLights = Array.from(document.querySelectorAll(this.data.sceneLights));
+    sceneLights.forEach(function (lightEl) {
+      var light = lightEl.getAttribute('light');
+      if (!light || light.intensity === undefined) {
+        return;
+      }
+      if (
+        lightEl === self.probeLight ||
+        lightEl === self.directionalLight
+      ) {
+        lightEl.components.light.light.intensity = 0;
+      } else {
+        var intensity = self.sceneLightsMap.get(lightEl);
+        if (intensity !== undefined) {
+          lightEl.components.light.light.intensity = intensity;
+        }
+      }
+    });
+
+    if (data.generateEnvironment) {
+      scene.environment = this.cubeRenderTarget.texture;
+      this.needsEnvironmentUpdate = true;
+    } else {
+      scene.environment = null;
+    }
+  },
+  startLightProbe: function () {
+    var data = this.data;
+    var scene = this.el.sceneEl.object3D;
+
+    if (data.generateEnvironment) {
+      this.needsLightProbeUpdate = true;
+      this.needsEnvironmentUpdate = true;
+    } else {
+      scene.environment = null;
+    }
   },
   update: function () {
     var scene = this.el.sceneEl.object3D;
     var data = this.data;
     var object3D = this.el.object3D;
+
     if (data.transparent) {
       object3D.background = null;
     } else {
       object3D.background = new THREE.Color(data.color);
     }
 
-    if (scene.environment && scene.environment !== this.cubeRenderTarget.texture) {
-      console.warn('Background will not overide predefined environment maps');
+    if (scene.environment &&
+      (scene.environment !== this.cubeRenderTarget.texture || scene.environment !== this.lightProbeTarget.texture)
+    ) {
+      console.warn('Background will not override user defined environment maps');
       return;
     }
 
     if (data.generateEnvironment) {
       scene.environment = this.cubeRenderTarget.texture;
+      this.needsEnvironmentUpdate = true;
     } else {
       scene.environment = null;
     }
+
+    if (this.updateInterval) {
+      clearInterval(this.updateInterval);
+    }
+    if (this.data.environmentUpdateFrequency > 0) {
```

Comment:
We can do this in tick with [delta](https://aframe.io/docs/1.2.0/core/component.html#tick-time-timedelta) instead of using setInterval